references/cli/addon.go

```diff
@@ -357,7 +358,69 @@ func NewAddonStatusCommand(c common.Args, ioStream cmdutil.IOStreams) *cobra.Com
 			return nil
 		},
 	}
-	cmd.Flags().BoolVarP(&verboseSatatus, "verbose", "v", false, "show addon descriptions and parameters in addition to status")
+	cmd.Flags().BoolVarP(&verboseStatus, "verbose", "v", false, "show addon descriptions and parameters in addition to status")
+	return cmd
+}
+
+// NewAddonCreateCommand creates an addon scaffold
+func NewAddonCreateCommand() *cobra.Command {
+	var helmRepoURL string
+	var chartName string
+	var chartVersion string
+	var path string
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "create an addon scaffold",
+		Long:  "Create an addon scaffold for quick starting. A Helm Component is generated if you provide Chart-related parameters.",
+		Example: `	vela addon init mongodb --helm-repo-url=https://marketplace.azurecr.io/helm/v1/repo --chart=mongodb --version=12.1.16
+will create something like this:
+	mongodb/
+	├── definitions
+	├── metadata.yaml
+	├── readme.md
+	├── resources
+	│   └── mongodb.cue
+	├── schemas
+	└── template.yaml
+
+If you want to store the scaffold in a different directory, you can use the -p/--path flag:
+	vela addon init mongodb -p ./some/repo --helm-repo-url=https://marketplace.azurecr.io/helm/v1/repo --chart=mongodb --version=12.1.16
+
+If you don't want the Helm component, just omit the three Chart-related parameters. We will create an empty scaffold for you.
+	vela addon init mongodb`,
```

Comment:
The empty scaffold should also create other addon dir for convenience, it's better to put some example files in it.

https://kubevela.io/docs/platform-engineers/addon/intro#build-an-addon

Reply:
Yes, there will be a minimal example in it (required files of an addon and other directories). 
<details>
<summary>File Structure and Contents (Basic)</summary>

```console
$ vela addon init mongo-sample
Scaffold created in directory mongo-sample. What to do next:
- Check out our guide on how to build your own addon: https://kubevela.io/docs/platform-engineers/addon/intro
- Review and edit what we have generated in mongo-sample
- To enable the addon, run: vela addon enable mongo-sample
```
```console
$ tree mongo-sample
mongo-sample
├── definitions
├── metadata.yaml
├── readme.md
├── resources
├── schemas
└── template.yaml

3 directories, 3 files
```
```console
$ cat mongo-sample/metadata.yaml
description: An addon for KubeVela.
icon: ""
invisible: false
name: mongo-sample
version: 1.0.0
```
```console
$ cat mongo-sample/readme.md
# mongo-sample

Also check how to build your own addon: https://kubevela.net/docs/platform-engineers/addon/intro

## Directory Structure

- `template.yaml`: contains the basic app, you can add some component and workflow to meet your requirements. Other files in `resources/` and `definitions/` will be rendered as Components and appended in `spec.components`
- `metadata.yaml`: contains addon metadata information.
- `definitions/`: contains the X-Definition yaml/cue files. These file will be rendered as KubeVela Component in `template.yaml`
- `resources/`:
  - `parameter.cue` to expose parameters. It will be converted to JSON schema and rendered in UI forms.
  - All other files will be rendered as KubeVela Components. It can be one of the two types:
    - YAML file that contains only one resource. This will be rendered as a `raw` component
    - CUE template file that can read user input as `parameter.XXX` as defined `parameter.cue`.
      Basically the CUE template file will be combined with `parameter.cue` to render a resource.
      **You can specify the type and trait in this format**
```
```console
$ cat mongo-sample/template.yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  creationTimestamp: null
  name: mongo-sample
  namespace: vela-system
spec:
  components: null
status: {}

```
</details>

Reply:
great